src/main/java/cd/go/contrib/elasticagent/KubernetesInstance.java

```diff
@@ -29,20 +29,22 @@
     private final Map<String, String> properties;
     private final Long jobId;
     private final PodState state;
+    private final ElasticProfileSettings settings;
 
-    public KubernetesInstance(DateTime createdAt, String environment, String name, Map<String, String> properties, Long jobId, PodState state) {
+    public KubernetesInstance(DateTime createdAt, ElasticProfileSettings settings,String environment, String name, Map<String, String> properties, Long jobId, PodState state) {
         this.createdAt = createdAt.withZone(DateTimeZone.UTC);
         this.environment = environment;
         this.name = name;
         this.properties = properties;
         this.jobId = jobId;
         this.state = state;
+        this.settings = settings;
```

Comment:
Duplicate Information.
`properties` and `settings` field will have duplicate information about `security-token`, `namespace` and `auto-register-timeout`.

Also, storing settings on the instance is a bad idea as it will contain stale information if setting has changed after the instance is created.

Reply:
The Properties here is the Pod Meta which does not have security-token and auto-register-timeout
.....
ObjectMeta metadata = elasticAgentPod.getMetadata();
......
 kubernetesInstance = new KubernetesInstance(createdAt, settings,environment, metadata.getName(), metadata.getAnnotations(), jobId, PodState.fromPod(elasticAgentPod));

Reply:
Also, I have a different opinion. Storing settings on the instance is storing the state of the Elastic Profile at the point in time when Pod was created. If the Namespace of Profile is modified later and the Pod is still Active, the setting at that point in time should be used to display the status in Elastic Profile Status report and for termination later on.
Otherwise the Pod can not be terminated if it was created in a different namespace and later on namespace was changed in Elastic Profile